### PR TITLE
chore: add cookieconsent-script & full-story / UG integration

### PR DESCRIPTION
**Pull request by misama-ct**
#### Summary
The cookie consent script, along with the FullStory and UserGuiding integrations, has been added to the project.

## Description
Storybook only allows adding HTML tags to the head section of the build. Unfortunately, there is no built-in way to differentiate between production and development builds, so I had to find a solution to handle this.

Here’s how the mechanism works:

- `manager-head.html` is Storybook’s default method for adding HTML tags to the head section of a deployment. This file remains unchanged.
- `manager-head.prod.html` contains content specific to the production version of Storybook. In production environments, the content of this file is appended to the regular `manager-head.html`. Additionally, any `{{VARIABLE}}` placeholders in the production file are replaced with corresponding environment variables.

> [!Note]  
> The UserGuiding (UG) and FullStory (FS) scripts have the attribute `type="text/plain" class="optanon-category-C0002"` added to their script tags.  If a user consents to performance cookies, these tags are targeted by the cookie consent script, rewritten, and the scripts are executed—at least in theory.  
> (Unfortunately, this does not work on Vercel preview domains because they are not on the *.commercetools.com domain.)

### CODE REVIEW

File: storybook/.storybook/manager-head.prod.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+(o=n.createElement(t)).async=!0,o.crossOrigin="anonymous",o.src="https://"+l,o.onerror=function(){y("Error loading "+l)}
+;var b=n.getElementsByTagName(t)[0];b&&b.parentNode?b.parentNode.insertBefore(o,b):n.head.appendChild(o)}(),function(){function b(){}
+function h(b,h,d){g(b,h,d,1)}function d(b,d,j){h("setProperties",{type:b,properties:d},j)}function j(b,h){d("user",b,h)}function r(b,h,d){j({
+uid:b},d),h&&j(h,d)}g.identify=r,g.setUserVars=j,g.identifyAccount=b,g.clearUserCookie=b,g.setVars=d,g.event=function(b,d,j){h("trackEvent",{
+name:b,properties:d},j)},g.anonymize=function(){r(!1)},g.shutdown=function(){h("shutdown")},g.restart=function(){h("restart")},
+g.log=function(b,d){h("log",{level:b,msg:d})},g.consent=function(b){h("setIdentity",{consent:!arguments.length||b})}}(),s="fetch",
+f="XMLHttpRequest",g._w={},g._w[f]=m[f],g._w[s]=m[s],m[s]&&(m[s]=function(){return g._w[s].apply(this,arguments)}),g._v="2.0.0")
+}(window,document,window._fs_namespace,"script",window._fs_script);
+</script>
+<!-- UserGudining -->
```
Comment (jaikamat) [inline storybook/.storybook/manager-head.prod.html:27]: nit: spelling, in case we have to find by keyword 😉

Comment (misama-ct) [reply storybook/.storybook/manager-head.prod.html:27]: Just testing you 💅

File: storybook/.storybook/manager-head.prod.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!-- Cookie Consent -->
+<script type="text/javascript" src="https://cdn.cookielaw.org/consent/{{COOKIELAW_ORG}}/OtAutoBlock.js"></script>
+<script type="text/javascript" src="https://cdn.cookielaw.org/scripttemplates/otSDKStub.js" data-domain-script="{{COOKIELAW_ORG}}"></script>
+<!-- FullStory -->
+<script type="text/plain" class="optanon-category-C0002">
```
Comment (jaikamat) [inline storybook/.storybook/manager-head.prod.html:5]: > The UserGuiding (UG) and FullStory (FS) scripts have the attribute type="text/plain" class="optanon-category-C0002" added to their script tags. If a user consents to performance cookies, these tags are targeted by the cookie consent script, rewritten, and the scripts are executed

Can we document this here? I fear this relationship might be lost down the line

**Review by ddouglasz — APPROVED** [review]
Thank you!

**Review by jaikamat — APPROVED** [review]
🏅

File: storybook/.storybook/manager-head.prod.html (hunk 2 above)
Comment (misama-ct) [reply storybook/.storybook/manager-head.prod.html:9]: I added a comment and a link. It's complicated though. We use cookielaw.org powered by cookierpro brought to you by onetrust.com 🫠

File: storybook/scripts/prepare-manager-head-file.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const envVars: Record<string, string | undefined> = {
+  /** ENV's needed for CookieConsent script integration */
+  COOKIELAW_ORG: process.env.COOKIELAW_ORG,
+  /** ENV's needed for FullStory integration */
+  FS_HOST: process.env.FS_HOST,
+  FS_SCRIPT: process.env.FS_SCRIPT,
+  FS_ORG: process.env.FS_ORG,
+  /** ENV's needed for UserGuiding integration */
+  USERGUIDING_ORG: process.env.USERGUIDING_ORG,
```
Comment (ByronDWall) [inline storybook/scripts/prepare-manager-head-file.ts:9]: are these env vars set in vercel?  When I build storybook for production and serve it, I get a 404 for this URL:
`https://cdn.cookielaw.org/consent//OtAutoBlock.js`, which I assume means that the `COOKIELAW_ORG` env var isnt set and this script is failing:

```<script type="text/javascript" src="https://cdn.cookielaw.org/consent/{{COOKIELAW_ORG}}/OtAutoBlock.js"></script>```

**Pull request merged**